script.js

```diff
@@ -259,4 +259,37 @@ form.addEventListener('submit', (e) => {
     span.textContent = 'Email should be in lowercase';
     error.appendChild(span);
   }
-});
\ No newline at end of file
+});
+
+// localstorage
+
+const x = {
+  name: '',
+  email: '',
+  text: '',
+};
+let y = '';
+function store() {
+  y = JSON.stringify(x);
+  localStorage.setItem('form', y);
+}
+form.elements.user_name.addEventListener('change', () => {
+  x.name = form.elements.user_name.value;
+  store();
+});
+
+form.elements.user_email.addEventListener('change', () => {
+  x.email = form.elements.user_email.value;
+  store();
+});
+form.elements.user_message.addEventListener('change', () => {
+  x.text = form.elements.user_message.value;
+  store();
+});
+
+if (localStorage.getItem('form')) {
```

Comment:
- [ OPTIONAL] I suggest that you should always be consistent in your programming style.  for example, if your choose to skip a line between functions then you should stick to that style throughout the code base. i therefore suggest that you  a blank line after line `284` and `275`. 😄